examples/release_highlights/plot_release_highlights_0_24_0.py

```diff
@@ -0,0 +1,213 @@
+# :class:`~sklearn.model_selection.HalvingGridSearchCV` and
+# :class:`~sklearn.model_selection.HalvingRandomSearchCV` can be
+# used as drop-in replacement for
+# :class:`~sklearn.model_selection.GridSearchCV` and
+# :class:`~sklearn.model_selection.RandomizedSearchCV`.
+# Successive halving is an iterative selection process. The first iteration is
+# run with a small amount of resources, resource meaning typically the
+# number of training samples, but also arbitrary integer parameters such
+# as `n_estimators` in a random forest. Only some of the parameter candidates
+# are selected for the next iteration, with an increasing size of the
+# allocated resources.
+# As illustrated in the figure below,
+# only a subset of candidates will last until the end of the iteration process.
+# Read more in the :ref:`User Guide <successive_halving_user_guide>`.
+# 
+# .. figure:: ../model_selection/images/sphx_glr_plot_successive_halving_iterations_001.png
+#   :target: ../model_selection/plot_successive_halving_iterations.html
+#   :align: center
+
+import numpy as np
+from scipy.stats import randint
+from sklearn.experimental import enable_halving_search_cv  # noqa
+from sklearn.model_selection import HalvingRandomSearchCV
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.datasets import make_classification
+
+rng = np.random.RandomState(0)
+
+X, y = make_classification(n_samples=700, random_state=rng)
+
+clf = RandomForestClassifier(n_estimators=20, random_state=rng)
+
+param_dist = {"max_depth": [3, None],
+              "max_features": randint(1, 11),
+              "min_samples_split": randint(2, 11),
+              "bootstrap": [True, False],
+              "criterion": ["gini", "entropy"]}
+
+rsh = HalvingRandomSearchCV(estimator=clf, param_distributions=param_dist,
+                            factor=2, random_state=rng)
+rsh.fit(X, y)
+rsh.best_params_
+
+##############################################################################
+# Native support for categorical features in HistGradientBoosting
+# ---------------------------------------------------------------
+# :class:`sklearn.ensemble.HistGradientBoostingClassifier` and
+# :class:`sklearn.ensemble.HistGradientBoostingRegressor` now have native
+# support for categorical features: they can consider splits on non-ordered,
+# categorical data. Read more in the :ref:`User Guide
+# <categorical_support_gbdt>`.
+#
+# .. figure:: ../ensemble/images/sphx_glr_plot_gradient_boosting_categorical_001.png
+#   :target: ../ensemble/plot_gradient_boosting_categorical.html
+#   :align: center
+
+##############################################################################
+# Improved performances in HistGradientBoosting methods
+# -----------------------------------------------------
+# Histogram initialization is now done in parallel in
+# :class:`ensemble.HistGradientBoostingRegressor` and
+# :class:`ensemble.HistGradientBoostingClassifier` which results in memory
```

Comment:
The memory footprint improvement doesn't come from histogram initialization. I'd suggest:

> The memory footprint of HistXX and HistYY was significantly improved during calls to `fit`. In addition, histogram initialization is now done in parallel which results in slight speed improvement.

It's important to note "during calls to fit" because the memory footprint of the overall fitted models wasn't changed at all.